tests/test_utilities.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import print_function
+
+import pytest
+
+import json
+from hashlib import sha256
+
+import flowpipe.utilities as util
+
+
+class WeirdObject(object):
+    """An object that is not json serializable and has no bytes() interface."""
+
+    foo = "bar"
+
+
+def test_node_encoder():
+    """Test the custom JSONEncoder."""
+    valid_object = {"key": "value", "other_key": [1, 2, 3]}
+    json_string = json.dumps(valid_object)
+    recovered_json = json.loads(json_string)
+    for k, v in valid_object.items():
+        assert v == recovered_json[k]
+
+    bytes_object = {"key": "value", "other_key": bytes(42)}
+    json_string = json.dumps(bytes_object, cls=util.NodeEncoder)
+    recovered_json = json.loads(json_string)
+    for k, v in bytes_object.items():
+        assert v == recovered_json[k] \
+            or sha256(v).hexdigest() == recovered_json[k]
+
+    weird_object = {"key": "value", "other_key": WeirdObject()}
+    json_string = json.dumps(weird_object, cls=util.NodeEncoder)
+    recovered_json = json.loads(json_string)
+    for k, v in bytes_object.items():
```

Comment:
I think the issue we get in the tests comes from a typo here, it should be: `for k, v in weird_object .items():`